test/libsolidity/SMTChecker.cpp

```diff
@@ -230,6 +230,21 @@ BOOST_AUTO_TEST_CASE(compound_assignment_division)
 	CHECK_WARNING(text, "Assertion violation");
 }
 
+BOOST_AUTO_TEST_CASE(mod)
+{
+	string text = R"(
+		contract C {
+			function f(int x, int y) public pure {
+				require(y < 0);
+				int z1 = x % y;
+				int z2 = x % -y;
+				assert(z1 == z2);
+			}
+		}
+	)";
+	CHECK_WARNING(text, "Overflow");
```

Comment:
What overflows here? Ah - probably ``-y`` for ``y = -2^255``, right.

Reply:
Yep, exactly the case you mentioned